src/components/image/index.tsx

```diff
@@ -52,6 +53,20 @@ function getContentApiDimensions(
 	return null
 }
 
+function getTheme(src: string): GlobalThemeOption | undefined {
+	let theme
+	const url = new URL(src, 'https://developer.hashicorp.com')
+	const themedImageSuffix = new RegExp(/#hide-on-(dark|light)/)
+
+	if (themedImageSuffix.test(url.hash)) {
+		const match = url.hash.match(themedImageSuffix)
+		// match capture group (dark|light)
+		theme = match[1]
```

Comment:
💭 I wonder if I should link this to global theme options.

Reply:
(optional) Wanna share an example of what you're hinting at here?

Reply:
oh yeah! [this enum](https://github.com/hashicorp/dev-portal/blob/72577b4f2f0b5a200e542c9110c0dfe345f5d997/src/styles/themes/types.ts#L6)

Reply:
I updated the return type, feels like enough for now...